store/sqlstore/channel_store.go

```diff
@@ -1805,25 +1805,30 @@ func (s SqlChannelStore) UpdateLastViewedAt(channelIds []string, userId string)
 	return times, nil
 }
 
-// countPostsAfter returns the number of posts in the given channel created after but not including the given timestamp.
-func (s SqlChannelStore) countPostsAfter(channelID string, since int64) (int64, *model.AppError) {
+// CountPostsAfter returns the number of posts in the given channel created after but not including the given timestamp. If given a non-empty user ID, only counts posts made by that user.
+func (s SqlChannelStore) CountPostsAfter(channelId string, timestamp int64, userId string) (int64, *model.AppError) {
 	countUnreadQuery := `
 	SELECT count(*)
 	FROM Posts
 	WHERE
-		ChannelId = :channelId
-		AND CreateAt > :createAt
-		AND Type = ''
+		ChannelId = :ChannelId
+		AND CreateAt > :CreateAt
+		AND Type = '' -- This line causes MM-20681
```

Comment:
I intentionally left this comment for myself since this ticket is also assigned to 5.18, and I'll be fixing it next.